internal/services/streamanalytics/stream_analytics_output_table_resource.go

```diff
@@ -236,6 +257,16 @@ func (r OutputTableResource) Update() sdk.ResourceFunc {
 				},
 			}
 
+			tableOutput, ok := props.OutputProperties.Datasource.AsAzureTableOutputDataSource()
+			if !ok {
+				return fmt.Errorf("converting output data source to a table output: %+v", err)
+			}
+			if v := state.ColumnsToRemove; v != nil && len(v) > 0 {
+				tableOutput.ColumnsToRemove = &v
+			} else {
+				tableOutput.ColumnsToRemove = &[]string{}
+			}
```

Comment:
This could be rewritten in a simplified manner so that the else condition isn't required anymore
```suggestion
			if metadata.ResourceData.HasChange("columns_to_remove") {
				tableOutput, ok := props.OutputProperties.Datasource.AsAzureTableOutputDataSource()
				if !ok {
					return fmt.Errorf("converting output data source to a table output: %+v", err)
				}
				tableOutput.ColumnsToRemove = &state.ColumnsToRemove
			}
```

Reply:
Updated